File: sbe/binance/spot_3_2/generated-python/spot_sbe/ErrorResponse.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple
from io import BytesIO

class ErrorResponse:
    """SBE message: ErrorResponse."""

    TEMPLATE_ID = 100
    SCHEMA_ID = 3
    SCHEMA_VERSION = 2
    BLOCK_LENGTH = 18

    def __init__(self):
        self.code = None
        self.server_time = None
        self.retry_after = None
        self.msg = b''
        self.data = b''

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.code is not None:
            buffer.write(struct.pack('<h', self.code))
        if self.server_time is not None:
            buffer.write(struct.pack('<q', self.server_time))
        if self.retry_after is not None:
            buffer.write(struct.pack('<q', self.retry_after))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        buffer = BytesIO(data)

        self.code = struct.unpack('<h', buffer.read(2))[0]
        self.server_time = struct.unpack('<q', buffer.read(8))[0]
        self.retry_after = struct.unpack('<q', buffer.read(8))[0]

File: sbe/binance/spot_3_2/generated-php/spot_sbe/ErrorResponse.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class ErrorResponse
{
    public const TEMPLATE_ID = 100;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 2;
    public const BLOCK_LENGTH = 18;

    public int|float|array|null $code = null;
    public int|float|array|null $serverTime = null;
    public int|float|array|null $retryAfter = null;
    public string $msg = '';
    public string $data = '';

    public function encode(): string
    {
        $buffer = '';

        if ($this->code !== null) {
            $buffer .= pack('s', $this->code);
        }
        if ($this->serverTime !== null) {
            $buffer .= pack('q', $this->serverTime);
        }
        if ($this->retryAfter !== null) {
            $buffer .= pack('q', $this->retryAfter);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->code = unpack('s', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->serverTime = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->retryAfter = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
    }
}
